bigquery_etl/cli/query.py

```diff
@@ -1262,6 +1265,151 @@ def render(name, sql_dir, output_dir):
             click.echo(rendered_sql)
 
 
+def _parse_partition_date(partition_date):
+    params = partition_date.split(":")
+    if len(params) != 3:
+        return None
+
+    # Check date format
+    try:
+        datetime.datetime.strptime(params[2], "%Y-%m-%d").date()
+    except ValueError:
+        return None
+
+    # Check column name
+    if re.match(r"^\w+$", params[0]):
+        return {params[0]: params[2]}
+
+
+def _validate_partition_date(ctx, param, partition_date):
+    """Process the CLI parameter check_date and set the parameter for BigQuery."""
+    # Will be None if launched from Airflow.  Also ctx.args is not populated at this stage.
+    if partition_date is None:
+        return None
+    else:
+        parsed = _parse_partition_date(partition_date)
+        if parsed is None:
+            raise click.BadParameter("Format must be <column-name>::<yyyy-mm-dd>")
+        return parsed
+
+
+def _parse_check_output(output: str) -> str:
+    output = output.replace("\n", " ")
+    if "ETL Data Check Failed:" in output:
+        return f"ETL Data Check Failed:{output.split('ETL Data Check Failed:')[1]}"
+    return output
+
+
+@query.command(
+    help="""
+    \b
+    UNDER ACTIVE DEVELOPMENT See https://mozilla-hub.atlassian.net/browse/DENG-919
+
+    Run ETL checks.
+    \b
+
+    Example:
+
+     ./bqetl query check ga_derived.downloads_with_attribution_v2 --partition download_date::2023-05-01
+    """,
+    context_settings=dict(
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--partition",
+    "-p",
+    help="Partition to check, format <column-name>::<yyy-mm-dd>, must be provided if not executing in Airflow ",
```

Comment:
Should we consider `column-name` in this case being a separate command param?

Reply:
We could but I wanted to keep the command param the same as the ctx.arg format/value so only one validation path is needed.

Reply:
I'm wondering if we should have a more generic `--parameters` option here instead. There are some queries which have additional parameters besides the partition date which could be specified then as well.